hoverkite-protocol/src/lib.rs

```diff
@@ -77,13 +80,52 @@ impl Command {
             }
             Command::Recenter => writer.bwrite_all(&[b'e'])?,
             Command::ReportBattery => writer.bwrite_all(&[b'b'])?,
+            Command::ReportCharger => writer.bwrite_all(&[b'c'])?,
             Command::RemoveTarget => writer.bwrite_all(&[b'n'])?,
             Command::PowerOff => writer.bwrite_all(&[b'p'])?,
         };
         Ok(())
     }
+
+    pub fn parse(buf: &[u8]) -> nb::Result<Self, ParseError> {
```

Comment:
It doesn't need to be fixed in this PR, but there's an implicit assumption here that parse either consumes the entire buffer, or nothing, and that this doesn't need to be checked because the caller will call it with a buffer size that increases by exactly 1 each time. We should make this explicit somehow, maybe be returning an error if the buffer is larger than the command expects.

Reply:
I started out trying to make these methods take a Read<u8> with some complicated SplitBuffer implementation, but then I got into trouble with Message::parse() needing to peek at the first byte in order to work out how to dispatch correctly. In the end, I decide to make this PR as small as minimal a refactor as possible.

I've added a note about this to the serde section of #5 though. It might fall out in the wash if serde checks for trailing bytes, and we use COBS/something for packetization.